.github/workflows/release.yml

```diff
@@ -21,33 +21,46 @@ jobs:
 
     steps:
       # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           repository: wingtk/gvsbuild
           fetch-depth: 0
 
       - name: Set up Python
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v3
         with:
           python-version: ${{ matrix.python }}
           architecture: ${{ matrix.arch }}
 
       - name: Install Build Deps
-        run: python -m pip install wheel
+        run: python -m pip install wheel build
+
+      - name: Install Poetry
+        run: python -m pip install poetry==1.1.13
+      - name: Configure Poetry
+        run: poetry config virtualenvs.in-project true
+      - name: Install Python Dependencies
+        run: poetry install
 
       - name: Move git binary
         # Temporarily move preinstalled git to prevent errors related to cygwin.
         run: Rename-Item "C:\Program Files\Git\usr\bin" notbin
 
+
+      - name: Pre-Build
+        run: 
+          pip install .
+
       - name: Build
         run: >
-          python build.py build
+          gvsbuild build
           --platform=${{ matrix.arch }}
           --vs-ver=${{ matrix.vstudio }}
           --same-python
           --enable-gi
           --py-wheel
-          --skip gtksourceview,emeus,clutter
+          --patches-root-dir=${{ github.workspace }}\patches
```

Comment:
What are these patches for?

Reply:
This is just the default patches folder that come with the repo, no idea what in there might be used or not it is just apparently not specified by default and fails if you dont. Likely something they should change to have a default but till then, its added.